sdk/core/azure-core/src/datetime.cpp

```diff
@@ -424,9 +424,13 @@ DateTime::operator std::chrono::system_clock::time_point() const
 
 DateTime DateTime::Parse(std::string const& dateTime, DateFormat format)
 {
-  int16_t year;
-  int8_t month;
-  int8_t day;
+  // The values that are not supposed to be read before they are written are set to -123... to avoid
+  // warnings on some compilers, yet provide a clearly bad value to make it obvious if things don't
+  // work as expected.
+  int16_t year = -12345;
+  int8_t month = -123;
+  int8_t day = -123;
```

Comment:
We should follow what was done below and set these to 0, rather than arbitrary negative values:
https://github.com/Azure/azure-sdk-for-cpp/blob/2cc9099e1ae63d9905ceceddee3d38aa1e991af0/sdk/core/azure-core/src/datetime.cpp#L434-L437

The risk of picking such sentinel values without a reason, is that overflow or other checks in the future that look for `year < 0` would start returning true.

(As an aside, if we really wanted to pick a random value, its arbitrary, and hence unexpected t pick different values just because the size of the integer type changed).

Also, initializing them to zero lets us remove the comment, which seem noisy if we were to add them everywhere we init locals.

Reply:
Also Q&A 4 here helps motivate the use of auto and declare at initialization to circumvent this problem all together:
https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/